### PR TITLE
[JavaScript] Add new concept exercise: classes-intro

### PR DESCRIPTION
**Pull request by rishiosaur**
This PR brings in the implementation described in #1325.

### CODE REVIEW

**Comment by rishiosaur** [comment]
NOTE: THIS IS STILL A DRAFT

**Comment by rishiosaur** [comment]
Things left to do: 

- [ ] Finish instructions
- [ ] Work on constructor and method tests

**Comment by rishiosaur** [comment]
Testing is all done. Now I just need to add in all the instructions

**Comment by ErikSchierboom** [comment]
@rishiosaur The main reviewer for the JavaScript track (@SleeplessByte) is temporarily unable to do reviews. It might take a little longer for this to be reviewed. Thanks for the patience!

**Comment by rishiosaur** [comment]
No worries @ErikSchierboom!

File: languages/javascript/exercises/concept/classes-intro/.docs/introduction.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+```js
+
+class MyClass() {
+    //...
+}
+
+```
+
+To add a method (a function of a class) to a class, just write the function definition without the `function` keyword in the class body:
```
Comment (pertrai1) [inline languages/javascript/exercises/concept/classes-intro/.docs/introduction.md:15]: I would remove the use of "just"

File: languages/javascript/exercises/concept/classes-intro/.docs/introduction.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+```js
+class FunctionDemo {
```
Comment (pertrai1) [inline languages/javascript/exercises/concept/classes-intro/.docs/introduction.md:18]: I would see this easier to understand without using the word 'function' in the class name and the methods.

File: languages/javascript/exercises/concept/classes-intro/.docs/introduction.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  constructor(property1, property2) {
+    this.property1 = property1
+    console.log(property2)
+  }
+}
+```
+
+The `this` keyword is a way to access properties and methods on a class's _prototype_. The prototype is a JavaScript object where all properties and methods of a class are stored.
+
+You can access the `this` keyword from anywhere inside any function:
```
Comment (pertrai1) [inline languages/javascript/exercises/concept/classes-intro/.docs/introduction.md:44]: I would make it clear that "any function" is "any function" that is on the class

File: languages/javascript/exercises/concept/classes-intro/.docs/introduction.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    // This line binds the value of the property1 argument to the property1 property on the class.
+    this.property1 = property1
+  }
+
+  useProperty() {
+    // We can use properties that are stored on the prototype anywhere
+    console.log(this.property1)
+  }
+
+  myFunction() {
```
Comment (pertrai1) [inline languages/javascript/exercises/concept/classes-intro/.docs/introduction.md:58]: Would remove the word **Function** here. Also, is this a time to discuss when to call it a function and when to call it a method?

**Comment by ErikSchierboom** [comment]
This Friday (Jan 29th 2021), we're going to explode this repo back into the individual track repositories and archive this repo.

As such, this PR needs to be merged by Friday 9am UTC, or it will need to be manually reopened against the track repo.

**Comment by SleeplessByte** [comment]
@rishiosaur if possible, can you rebase? If not I can do it later.

**Pull request closed without merging**